Python/sysmodule.c

```diff
@@ -2180,6 +2180,38 @@ sys_is_stack_trampoline_active_impl(PyObject *module)
 }
 
 
+/*[clinic input]
+sys._get_calling_module_name
+
+    depth: int = 0
+
+Return the name of the calling module.
+
+The default depth returns the module containing the call to this function.
+A more typical use in a library will pass a depth of 1 to get the user's
+module rather than the library module.
+[clinic start generated code]*/
+
+static PyObject *
+sys__get_calling_module_name_impl(PyObject *module, int depth)
+/*[clinic end generated code: output=bd04c211226f8b84 input=dd268ae6a20311ab]*/
+{
+    _PyInterpreterFrame *f = _PyThreadState_GET()->cframe->current_frame;
+    while (f && (f->owner == FRAME_OWNED_BY_CSTACK || depth-- > 0)) {
+        f = f->previous;
+    }
```

Comment:
@markshannon, could you confirm that this is how I should be traversing up the call stack now? The equivalent of `getframe().f_back.f_back...`.

Right now I think I'm skipping over native frames, which seems to at least handle the cases I've designed for, but it's not clear to me whether they ought to be counted in `depth` or ignored and only checked after recursing.

Reply:
@markshannon re-ping, since I think you're back from holiday now right?

Reply:
Mark doesn't always receive reliable notifications from GitHub (I suspect there just are too many). But you can reliably summon him via Discord (or perhaps Teams :-).